sympy/solvers/solveset.py

```diff
@@ -3239,6 +3241,15 @@ def _solve_using_known_values(result, solver):
                         soln = solver(eq2, sym)
                         total_solvest_call += 1
                         soln_new = S.EmptySet
+                        if isinstance(soln, ConditionSet):
+                            if soln.args[2] in (S.Reals, S.Complexes):
```

Comment:
```suggestion
                            if soln.base_set in (S.Reals, S.Complexes):
```

But I'm not sure why pure Real/Complex are being tested since the `base_set` can be any set, e.g.
```python
>>> ConditionSet(x, x>3, Interval(0,5)).base_set
Interval(0, 5)
```

Reply:
> But I'm not sure why pure Real/Complex are being tested since the base_set can be any set, e.g.


Since nonlinsolve works with only S.Reals and S.Complexes so I've considered these 2 cases only. (please see https://github.com/sympy/sympy/pull/21212#discussion_r606552603)